docs/frontend-system/building-plugins/01-index.md

```diff
@@ -6,50 +6,153 @@ sidebar_label: Overview
 description: Building frontend plugins using the new frontend system
 ---
 
-> **NOTE: The new frontend system is in alpha and is only supported by a small number of plugins.**
+:::info
+The new frontend system is in alpha and only a few number of plugins have been migrated. The new frontend
+plugins are not backwards compatible with the old `createApp()`. As a result you will need to migrate your entire
+application to the new frontend system in order to use any of the new plugin functionality. This includes
+migrating your usage of `createApp()` to the new package `@backstage/frontend-defaults` which is still incomplete.
+
+To clarify, you cannot use the new `createFrontendPlugin()` unless you have migrated your entire application to
+the new frontend system.
+:::
+
+This section covers how to build frontend [plugins](../architecture/15-plugins.md). Plugins are one type of frontend _features_.
+Features, which include plugins and modules, are stitched together to build up a Backstage frontend [app](../architecture/10-app.md).
+
+Every plugin can incorporate one or more extensions. Extensions in the new frontend system are often generated from an
+`ExtensionBlueprint`. Extension Blueprints are opinionated factories for building extensions (e.g. `ApiBlueprint` is
+an `ExtensionBlueprint`).
+
+Putting it all together, a plugin in the new frontend system will:
+
+- Export a single `FrontendPlugin` object
+- Export one or more `ExtensionDefinition` objects, usually generated by an `ExtensionBlueprint`
```

Comment:
I don't think that this is true, `ExtensionDefinitions` shouldn't be exported from the package. Extensions should be wrapped up in a `Plugin` or `Module`.

Maybe we should change the above line to `Export a single FrontendFeature` or something instead?

Reply:
Your comments are really revealing on how easily it was for me to misunderstand what was going on :D One thing I might mention is that I am trying to deduce the way things work by both reading code and reading the way others have implemented the code. IIRC I deduced that you export an `ExtensionDefinition` object from an example somewhere, but I cannot recall where. I'll try to dig into this a bit today.